admin/src/components/DashboardModule.jsx

```diff
@@ -0,0 +1,58 @@
+import Switch from '../elements/Switch';
+import { setModule } from '../api/modules';
+import { ReactComponent as ArrowIcon } from "../assets/images/icon-arrow.svg";
+import { ReactComponent as ApiIcon } from "../assets/images/api-exclamation.svg";
+import "../assets/styles/components/_DashboardModule.scss";
+import { useState } from 'react';
+
+export default function DashboardModule({ moduleId, image, isActive, title, hasApi, children }) {
+	const [moduleActive, setModuleActive] = useState(isActive ? true : false);
+	const [activating, setIsActivating] = useState(false);
+	const handleSwitch = () => {
+		setIsActivating(true)
+		setModule(moduleId, { active: !moduleActive }).then(data => {
+			if (data) {
+				setModuleActive(data.active)
+				setIsActivating(false)
+				console.log(data)
+			}
+		}
+		);
+	}
+
+	return (
+		<div className={`urlslab-dashboardmodule ${activating ? 'activating' : ''} ${moduleActive ? 'active' : ''}`}>
+			{hasApi
+				? <div className="urlslab-dashboardmodule-api">
+					<ApiIcon />
+					This module requires API key
```

Comment:
change text to shorter version with option to translate: __('Urlslab API key required')